pkg/services/ngalert/services/persist.go

```diff
@@ -0,0 +1,24 @@
+package services
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+// AMStore is a store of Alertmanager configurations.
+type AMConfigStore interface {
+	GetLatestAlertmanagerConfiguration(ctx context.Context, query *models.GetLatestAlertmanagerConfigurationQuery) error
+	SaveAlertmanagerConfiguration(ctx context.Context, cmd *models.SaveAlertmanagerConfigurationCmd) error
```

Comment:
I think the function that we use here is the updated one, using the hash, etc. Save is the old one without concurrency guarantees.

Reply:
Done. I had to modify it slightly to propagate contexts so that transactions would work correctly.